# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GC.cpp
# Commit: 410d1b57bb09
# Full Hash: 410d1b57bb0955e0fe2f47b95ffa5ee1005e6791
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-09 21:37:17
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 5: Replace heap state checks that rely on there being a JSContext for the thread r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D140192
# ==============================================================================

diff -r f78d139e1d0e -r 410d1b57bb09 js/src/gc/GC.cpp
--- a/js/src/gc/GC.cpp	Wed Mar 09 10:17:07 2022 +0000
+++ b/js/src/gc/GC.cpp	Wed Mar 09 10:17:07 2022 +0000
@@ -1570,14 +1570,11 @@
                                           const HeapThreshold& threshold,
                                           JS::GCReason reason) {
   // Ignore malloc during sweeping, for example when we resize hash tables.
-  if (!CurrentThreadCanAccessRuntime(rt)) {
-    MOZ_ASSERT(JS::RuntimeHeapIsBusy());
+  if (heapState() != JS::HeapState::Idle) {
     return false;
   }
 
-  if (rt->heapState() != JS::HeapState::Idle) {
-    return false;
-  }
+  MOZ_ASSERT(CurrentThreadCanAccessRuntime(rt));
 
   TriggerResult trigger = checkHeapThreshold(zone, heap, threshold);
   if (!trigger.shouldTrigger) {